src/gt4py/next/ffront/ast_passes/single_static_assign.py

```diff
@@ -82,11 +179,78 @@ def apply(cls, node: ast.AST) -> ast.AST:
 
     def __init__(self, separator="__"):
```

Comment:
Some type annotations are missing. Additionally, I don't like the default separator since it may easily collide with user written code. I would suggest to use a very uncommon unicode symbol as separator to avoid problems (see below for further information about accepted unicode chars in python)
```suggestion
    def __init__(self, separator: str="ᐞ") -> None:
```
This would produce: `nameᐞ2`
```
Unicode symbols per class: https://www.compart.com/en/unicode/category
Python unicode symbols for identifiers:
    - xid_start  =  all characters in id_start after NFKC normalization, that is general categories Lu, Ll, Lt, Lm, Lo, Nl, and the underscore
    - xid_continue  =  all characters in id_continue after NFKC normalization, that is all chars in id_start plus characters in the categories Mn, Mc, Nd, Pc 

https://www.compart.com/en/unicode/U+10307: 𐌇
https://www.compart.com/en/unicode/U+10318: 𐌘
https://www.compart.com/en/unicode/U+A788: ꞈ
https://www.compart.com/en/unicode/U+141E: ᐞ
https://www.compart.com/en/unicode/U+1421: ᐡ
https://www.compart.com/en/unicode/U+1506: ᔆ
https://www.compart.com/en/unicode/U+15B0: ᖰ
https://www.compart.com/en/unicode/U+15B1: ᖱ
https://www.compart.com/en/unicode/U+15B2: ᖲ
https://www.compart.com/en/unicode/U+15B3: ᖳ
https://www.compart.com/en/unicode/U+1601: ᘁ
https://www.compart.com/en/unicode/U+166E: ᙮
https://www.compart.com/en/unicode/U+167F: ᙿ
```